### ISSUE TITLE
Example scripts in translation task process train/test sets in different ways.

### ISSUE THREAD

**Issue by pluiez** [pytorch/fairseq#1740]
#### What is your question?

Example [wmt14en2fr script](https://github.com/pytorch/fairseq/blob/master/examples/translation/prepare-wmt14en2fr.sh) applys `normalize-punctuation.perl`, `remove-non-printing-char.perl` and tokenization to process training set, whereas only tokenization is performed on valid/test set.

I'm wondering if this is a bug, or why not process test sets in the same way as train set?

#### Code

Code snippet to process training set:
```shell
echo "pre-processing train data..."
for l in $src $tgt; do
    rm $tmp/train.tags.$lang.tok.$l
    for f in "${CORPORA[@]}"; do
        cat $orig/$f.$l | \
            perl $NORM_PUNC $l | \
            perl $REM_NON_PRINT_CHAR | \
            perl $TOKENIZER -threads 8 -a -l $l >> $tmp/train.tags.$lang.tok.$l
    done
done
```

Code snippet to process test set:
```shell
echo "pre-processing test data..."
for l in $src $tgt; do
    if [ "$l" == "$src" ]; then
        t="src"
    else
        t="ref"
    fi
    grep '<seg id' $orig/test-full/newstest2014-fren-$t.$l.sgm | \
        sed -e 's/<seg id="[0-9]*">\s*//g' | \
        sed -e 's/\s*<\/seg>\s*//g' | \
        sed -e "s/\’/\'/g" | \
    perl $TOKENIZER -threads 8 -a -l $l > $tmp/test.$l
    echo ""
done
```

**Comment by myleott** [comment]
It's intentional, we try to limit the manipulation of the valid/test sets as much as possible. I've experimented with applying normalization to the test sets and it doesn't change the results much though (0.1 BLEU).
